pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -617,14 +637,55 @@ func removeFinalizerFromPVC(ctx context.Context, client client.Client, pvc *v1.P
 			pvc.Finalizers = append(pvc.Finalizers[:i], pvc.Finalizers[i+1:]...)
 		}
 	}
-	err := client.Update(ctx, pvc)
+	err := updateSVPVC(ctx, client, pvc)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debugf("PersistentVolumeClaim: %q on namespace: %q not found. Returning nil", pvc.Name, pvc.Namespace)
+			return nil
+		}
 		log.Errorf("failed to update PersistentVolumeClaim: %q on namespace: %q. Error: %+v",
 			pvc.Name, pvc.Namespace, err)
 	}
 	return err
 }
 
+func updateSVPVC(ctx context.Context, client client.Client,
```

Comment:
This method will be similar to `updateCnsNodeVMAttachment` method which reapplies the changes on the latest instance